### PR TITLE
Feature/issue 19 change AvroMessageUtil to generic RecordPostProcessor

### LINKED ISSUE #19: Replace AvroMessageUtil with Generic record PostProcessor
AvroMessageUtil does processing on records, making sure all fields are valid (part of that is setting nonnullable fields to valid values), before they are utilized in the TopologyTester. 

Should make an interface RecordPostProcessor that the AvroMessageUtil (renamed as AvroMessageProcessor) implements, allowing implementation of other processors for different types of messages (like protobuf)

### PR DESCRIPTION
**Pull request by GalateaWade**

### CODE REVIEW

File: kafka-starter-testing/src/main/java/org/galatea/kafka/starter/testing/TopologyTester.java [hunk 1]
```diff
@@ -56,12 +54,10 @@
   private final Map<String, TopicConfig<?, ?>> outputTopicConfig = new HashMap<>();
   private final Map<String, TopicConfig<?, ?>> storeConfig = new HashMap<>();
   private final Set<Class<?>> beanClasses = new HashSet<>();
-  private final Set<Class<?>> avroClasses = new HashSet<>();
 
   @Getter
   private final ConversionUtil typeConversionUtil = new ConversionUtil();
-  @Getter
-  private final AvroMessageUtil avroMessageUtil = AvroMessageUtil.defaultUtil();
+  private final Map<Class<?>, RecordPostProcessor<?>> postProcessors = new HashMap<>();
```
Comment (gala-emile) [inline kafka-starter-testing/src/main/java/org/galatea/kafka/starter/testing/TopologyTester.java:60]: Use `Maps.newHashMap()` to be in with the cool kids 😎

File: kafka-starter-testing/src/main/java/org/galatea/kafka/starter/testing/TopologyTester.java [hunk 2]
```diff
@@ -289,7 +279,8 @@ public void beforeTest() {
       return expectedOutput;
     }
 
-    Set<String> expectedFields = expectedRecordMaps.iterator().next().keySet();
+    Set<String> expectedFields = expectedRecordMaps.stream().findFirst().map(Map::keySet)
+        .orElse(new HashSet<>());
```
Comment (gala-emile) [inline kafka-starter-testing/src/main/java/org/galatea/kafka/starter/testing/TopologyTester.java:283]: `Sets.newHashset()` (it's really minor)

File: kafka-starter-testing/src/main/java/org/galatea/kafka/starter/testing/avro/AvroPostProcessor.java [hunk 3]
```diff
@@ -95,8 +103,8 @@ public void registerType(LogicalType type, AvroFieldType fieldType) {
       }
       message.put(field.pos(), fieldType.defaultValue());
     } else if (primitiveType.equals(Type.RECORD)) {
```
Comment (gala-emile) [inline kafka-starter-testing/src/main/java/org/galatea/kafka/starter/testing/avro/AvroPostProcessor.java:105]: Won't this create a NPE? You can't do `<null>.equals()`, right?? Pretty sure you can use `==` for enums.

**Comment by gala-emile** [comment]
Also fix conflicts

File: kafka-starter-testing/src/main/java/org/galatea/kafka/starter/testing/TopologyTester.java (hunk 1 above)
Comment (GalateaWade) [reply kafka-starter-testing/src/main/java/org/galatea/kafka/starter/testing/TopologyTester.java:60]: ehh, I see that as a helper method that's no longer useful, with the introduction of <> in Java8. The only difference is that maps.newHashMap() requires additional library.

File: kafka-starter-testing/src/main/java/org/galatea/kafka/starter/testing/avro/AvroPostProcessor.java (hunk 3 above)
Comment (GalateaWade) [reply kafka-starter-testing/src/main/java/org/galatea/kafka/starter/testing/avro/AvroPostProcessor.java:105]: I don't see how primitiveType could ever be null though. I think you're right we could use `==`, but I've gotten in the habit of using .equals() whenever possible

File: kafka-starter-testing/src/main/java/org/galatea/kafka/starter/testing/TopologyTester.java (hunk 1 above)
Comment (gala-emile) [reply kafka-starter-testing/src/main/java/org/galatea/kafka/starter/testing/TopologyTester.java:60]: Ahh ok, I just find it a bit neater -- personal preference dw

**Review by gala-emile — APPROVED** [review]

**Pull request merged**